### DISCUSSION TITLE
How to avoid multiple api calls in server components `generateMetadata` and default page components

### DISCUSSION THREAD

**Discussion by udayshankark-qp** [vercel/next.js#70312] [Help]
How can we avoid multiple api calls in server component such as page.tsx where generateMetaData and default Page components both using same api to fetch screen data and meta data

**Comment by Mr-Meshky** [comment]
Maybe this can help you
` { cache: 'force-cache' }`
[ Doc](https://nextjs.org/docs/app/api-reference/functions/fetch)

**Reply by udayshankark-qp** [reply]
@Mr-Meshky , this is already available, but then the routes are still quite slow because of the two api call

**Reply by Mr-Meshky** [reply]
Are you using pages route, right?

**Reply by udayshankark-qp** [reply]
@Mr-Meshky  We are using App route

**Reply by Mr-Meshky** [reply]
If you put a meta tag in the layout, can't you use it for all routes? Because what you said, they are all an api

**Reply by udayshankark-qp** [reply]
@Mr-Meshky what I meant was the page meta changes for each route.
The api to fetch meta information and the screen data is same for each route , since its api has dynamic parameters so the data wilo be different for each route

**Comment by Juneezee** [comment]
If you are using `fetch`, then Next.js does this for you automatically by default. See https://nextjs.org/docs/app/building-your-application/caching#request-memoization

**Reply by udayshankark-qp** [reply]
@Juneezee  you are right, but for some reason its taking lot of time to route from one page to another because of these 2 api call even though its same
